docs/modules/ROOT/pages/explanations/jsonnet.adoc

```diff
@@ -0,0 +1,91 @@
+= Jsonnet Best Practices
+
+== Merging External Config Data
+////
+* Deep merge helper ✔
+* Merge from hiera ✔
+* Expose defaults in hiera ✔
+* Dict with key=name (too keep them mergeable) ✔
+* Provide "generic" defaults and merge with context specifics (i.e. cloud/region see openshift4-nodes) ✔
+* Allow Null (to enable empty) ✔
+* For non-mergeable data (i.e. strings), provide defaults + extras ✔
+////
+If a component manages an external configuration it's considered best practice to expose this configuration in the hierarchy and to define a sensible default where possible.
+This allows to override the configuration via the hierarchy.
```

Comment:
Is the term _the hierarchy_ defined well enough?

Reply:
I added it to the glossary.